modules/apps/27-interchain-accounts/keeper/account.go

```diff
@@ -40,23 +40,28 @@ func (k Keeper) InitInterchainAccount(ctx sdk.Context, connectionID, counterpart
 	return nil
 }
 
-// Register interchain account if it has not already been created
-func (k Keeper) RegisterInterchainAccount(ctx sdk.Context, portID string) {
-	address := types.GenerateAddress(portID)
+// RegisterInterchainAccount attempts to create a new account using the provided address and stores it in state keyed by the provided port identifier
+// If an account for the provided address already exists this function returns nil
+func (k Keeper) RegisterInterchainAccount(ctx sdk.Context, address, portID string) error {
+	accAddr, err := sdk.AccAddressFromBech32(address)
+	if err != nil {
+		return err
+	}
 
-	account := k.accountKeeper.GetAccount(ctx, address)
-	if account != nil {
-		return
+	if acc := k.accountKeeper.GetAccount(ctx, accAddr); acc != nil {
+		return nil
```

Comment:
We should probably return a more descriptive error here. wdyt?

Reply:
Also, wondering if this check to get an account needs to change at all given we are using the new module acc feature?

Reply:
I think the point is to do a no-op if the account already exists since it could create an attack vector if an error is returned. Or at least that's what comes to mind. It'd be great to have an inline comment explaining the reasoning

Reply:
Is the idea here that there may be an account that already exists and this would return nil (no-op) in the event that a new channel was being established between chains that had already registered an interchain account?